packages/core/src/auth/client/accessToken.ts

```diff
@@ -0,0 +1,133 @@
+import { headlessConfig } from '../../config';
+import { isServerSide } from '../../utils';
+import isNil from 'lodash/isNil';
+import isString from 'lodash/isString';
+
+export interface AccessToken {
+  token: string | undefined;
+  expiration: number | undefined;
+}
+
+let accessToken: AccessToken | undefined;
+
+/**
+ * Get an access token from memory if one exists
+ *
+ * @returns {string | undefined}
+ */
+export function getAccessToken(): string | undefined {
+  return accessToken?.token;
+}
+
+/**
+ * Get an access token expiration from memory if one exists
+ *
+ * @returns {number | undefined}
+ */
+export function getAccessTokenExpiration(): number | undefined {
+  return accessToken?.expiration;
+}
+
+/**
+ * Set an access token and/or its expiration in memory
+ *
+ * @param {string} token
+ * @param {number} expiration
+ *
+ * @returns {void}
+ */
+export function setAccessToken(
+  token: string | undefined,
+  expiration: number | undefined,
+): void {
+  if (isServerSide()) {
+    return;
+  }
+
+  accessToken = {
+    token,
+    expiration,
+  };
+}
+
+/**
+ * Fetch an access token from the authorizeHandler middleware
+ *
+ * @export
+ * @param {string} code An authorization code to fetch an access token
+ */
+export async function fetchAccessToken(code?: string): Promise<string | null> {
+  const { apiEndpoint } = headlessConfig();
+
+  if (isNil(apiEndpoint)) {
+    throw new Error(
+      'You must provide an apiEndpoint value in your Headless config in order to use the fetchToken middleware',
+    );
+  }
+
+  let url = apiEndpoint;
+
+  // Add the code to the url if it exists
+  if (isString(code) && code.length > 0) {
+    url += `?code=${code}`;
+  }
+
+  try {
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    const result = (await response.json()) as {
+      accessToken: string;
+      accessTokenExpiration: number;
+    };
+
+    // If the response is not ok, clear the access token
+    if (!response.ok) {
+      setAccessToken(undefined, undefined);
+      return null;
+    }
+
+    setAccessToken(result.accessToken, result.accessTokenExpiration);
+
+    return result.accessToken;
+  } catch (error) {
+    setAccessToken(undefined, undefined);
+
+    return null;
+  }
+}
+
+/**
+ * The interval in which the access token is check if a new one
+ * needs to be fetched
+ */
+export const ACCESS_TOKEN_EXP_CHECK_INTERVAL = 15000;
```

Comment:
I believe this was my naming originally, but maybe we should call this `ACCESS_TOKEN_EXP_CHECK_INTERVAL_MS` since all other constants are in seconds.